### PR TITLE
Add retry_after option

### PR DESCRIPTION
**Pull request by Ninir**
This adds the retryAfter option.
I'm from another development language world, so tell me if I miss something :)

Should help for #91.

### CODE REVIEW

**Comment by Ninir** [comment]
@dadoonet This is a draft of the option addition. Open for reviews, remarks, etc... :)

File: src/main/java/org/elasticsearch/river/twitter/TwitterRiver.java [hunk 1]
```diff
@@ -62,6 +62,8 @@
     private final String oauthAccessToken;
     private final String oauthAccessTokenSecret;
 
+    private final int retryAfter;
```
Comment (dadoonet) [inline src/main/java/org/elasticsearch/river/twitter/TwitterRiver.java]: Better using `private final TimeValue retryAfter;`

Then you can set it using:

```java
this.retryAfter = XContentMapValues.nodeTimeValue(twitterSettings.get("retry_after"), TimeValue.timeValueSeconds(1));

// Or if it does not work something like

this.retryAfter = TimeValue.parseTimeValue(XContentMapValues.nodeStringValue(
                    twitterSettings.get("retry_after"), "1s"), TimeValue.timeValueSeconds(1));
```

**Comment by dadoonet** [comment]
Nice effort! :)

Left a comment. Could you look at tests and add a test just to ensure that the new parameter can be read when starting a river instance? 

Something like:

```java
    @Test
    public void testSample() throws IOException, InterruptedException {
        launchTest(jsonBuilder()
            .startObject()
                .field("type", "twitter")
                .startObject("twitter")
                    .field("type", "sample")
                    .field("retry_after", "10s")
               .endObject()
            .endObject(), randomIntBetween(10, 200), true);
    }
```

Thanks!

**Comment by Ninir** [comment]
@dadoonet Updated the code. However, I can't run the tests at the moment, compilation is fine, but the suite is failing with a big stack trace:

```
> Throwable #3: java.security.AccessControlException: access denied ("java.io.FilePermission" "/tmp/elasticsearch/myname.local-CHILD_VM=[0]-CLUSTER_SEED=[3948851719878273470]-HASH=[10FE0160A07BD]/nodes/0" "write")
```

Added all the permissions possible to this, but can't manage it :/

**Comment by dadoonet** [comment]
w00t. Just hit the same issue for another plugin. Going to check.

File: src/main/java/org/elasticsearch/river/twitter/TwitterRiver.java [hunk 2]
```diff
@@ -135,6 +137,12 @@ public TwitterRiver(RiverName riverName, RiverSettings riverSettings, Client cli
                 oauthAccessTokenSecret = settings.get("river.twitter.oauth.access_token_secret");
             }
 
+            if (twitterSettings.containsKey("retry_after")) {
+                retryAfter = XContentMapValues.nodeTimeValue(twitterSettings.get("retry_after"), TimeValue.timeValueSeconds(10));
+            } else {
+                retryAfter = TimeValue.timeValueSeconds(Long.parseLong(settings.get("river.twitter.retry_after")));
```
Comment (dadoonet) [inline src/main/java/org/elasticsearch/river/twitter/TwitterRiver.java]: We should in any case provide a default value: 

```java
retryAfter = XContentMapValues.nodeTimeValue(settings.get("river.twitter.retry_after"), TimeValue.timeValueSeconds(10));
```

File: src/main/java/org/elasticsearch/river/twitter/TwitterRiver.java [hunk 3]
```diff
@@ -293,6 +301,7 @@ public TwitterRiver(RiverName riverName, RiverSettings riverSettings, Client cli
             oauthConsumerSecret = settings.get("river.twitter.oauth.consumer_secret");
             oauthAccessToken = settings.get("river.twitter.oauth.access_token");
             oauthAccessTokenSecret = settings.get("river.twitter.oauth.access_token_secret");
+            retryAfter = TimeValue.timeValueSeconds(Long.parseLong(settings.get("river.twitter.retry_after")));
```
Comment (dadoonet) [inline src/main/java/org/elasticsearch/river/twitter/TwitterRiver.java]: We should in any case provide a default value: 

```java
retryAfter = XContentMapValues.nodeTimeValue(settings.get("river.twitter.retry_after"), TimeValue.timeValueSeconds(10));
```

**Comment by dadoonet** [comment]
I did a final review and left a comment.

Could you update your PR, squash all the commits and force push to your repo so I can merge it?

**Comment by Ninir** [comment]
@dadoonet Just done it! However, tests are still failing for the Exception I wrote earlier :/ any hints?

**Comment by dadoonet** [comment]
Actually tests looks good on my end but `testgeoAsArray` which is not caused by your commit IMO. Will merge your changes (and fix the error :) )

**Comment by dadoonet** [comment]
Sorry. One thing again. Could you add documentation in the README about this new setting?

**Comment by dadoonet** [comment]
@Ninir Sounds like you did not sign the CLA as well. Could you check it please?

**Comment by Ninir** [comment]
@dadoonet Just did it!

**Comment by Ninir** [comment]
@dadoonet Done. Do you find it pertinent and concise enough?

**Comment by dadoonet** [comment]
Great. The only bad thing is that it seems your IDE formatted the README so more than the simple changes you did have been added to the diff. Do you think you can fix that? If not, I can do it on my side.

**Comment by Ninir** [comment]
@dadoonet Done!

**Comment by dadoonet** [comment]
Merged in master with 770964a593e14be5ed1ab9294a975e768fcb4755, in 1.x with c01b04a0881c5551873ca74cdfad6c33bfbbcd72 and in 1.4 with 2705bfc25530c932f4d12b1f004939ab6acddfe8.

Thanks! :)

**Pull request closed without merging**